aplib.net-demo/Assets/Scripts/Entities/Weapons/RangedWeapon.cs

```diff
@@ -111,5 +111,10 @@ public bool EnemiesInLineOfSight()
 
             return firstHit.collider != null && firstHit.collider.CompareTag(_targetTag);
         }
+
+        /// <summary>
+        /// Check if the weapon has enough ammunition to be used and thus animated.
+        /// </summary>
+        public override bool CanAnimate() => !_ammoPouch.IsEmpty();
```

Comment:
I would rename this to something like "CanFire", so it can be used in more general scenarios